Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -138,9 +138,173 @@ public class PushDeploymentController : ApiController
             }
         }
 
-        private string GetZipURLFromARMJSON(JObject requestObject)
+        [HttpPost]
+        [HttpPut]
+        public async Task<HttpResponseMessage> OnePushDeploy(
+            [FromUri] string type = null,
+            [FromUri] bool async = false,
+            [FromUri] string path = null,
+            [FromUri] bool clean = false,
+            [FromUri] string message = DefaultMessage)
+        {
+            using (_tracer.Step("OnePushDeploy"))
+            {
+                var response = Request.CreateResponse();
+
+                var website_framework = System.Environment.GetEnvironmentVariable(Constants.Framework).ToLower();
+
+                var deploymentInfo = new ArtifactDeploymentInfo(_environment, _traceFactory)
+                {
+                    AllowDeploymentWhileScmDisabled = true,
+                    Deployer = Constants.OneDeploy,
+                    IsContinuous = false,
+                    AllowDeferredDeployment = false,
+                    IsReusable = false,
+                    CleanupTargetDirectory = clean,
+                    TargetChangeset = DeploymentManager.CreateTemporaryChangeSet(message: "Deploying from pushed file"),
+                    CommitId = null,
+                    RepositoryType = RepositoryType.None,
+                    Fetch = ArtifactFetch,
+                    DoFullBuildByDefault = false,
+                    Message = message,
+                    WatchedFileEnabled = false 
+                };
+
+                switch (type)
+                {
+                    case Constants.War:
+                        if(website_framework!=Constants.Tomcat)
+                        {
+                           
+                            response.StatusCode = HttpStatusCode.BadRequest;
```

Comment:
return Request.CreateResponse(HttpStatusCode.BadRequest, $"WAR files cannot be deployed to {website_framework}"; 

Will this work? Same comment elsewhere.